spec/queries/articles/active_threads_query_spec.rb

```diff
@@ -1,10 +1,6 @@
 require "rails_helper"
 
 RSpec.describe Articles::ActiveThreadsQuery, type: :query do
-  before do
-    create(:article, score: described_class::MINIMUM_SCORE - 1, tags: "watercooler")
-  end
```

Comment:
FYI: This line serves as a way to ensure that _only_ the correct articles are returned in the arrays throughout the spec. That said, there are two tests that fail when this line is included in the spec. By removing this line, the spec passes, which feels wrong and manufactured. I'm looking into this still, but can't reproduce this behavior (a post with a tag not belonging to the sidebar tag showing up beneath a sidebar tag/in the array) locally and cannot get the spec to pass yet.

Reply:
You don't want to remove this. Your specs failing actually indicate that we did break previous behavior. +1 for specs doing their job 🎉  
```
1) Articles::ActiveThreadsQuery::call when time_ago is latest returns any article if no higher-quality article is found
     Failure/Error: expect(result.length).to eq(1)
       expected: 1
            got: 2
       (compared using ==)
     # ./spec/queries/articles/active_threads_query_spec.rb:23:in `block (4 levels) in <main>'
```
That spec failed bc when we search for latest WITH a tag now we get 2 articles back, one of which DOES NOT have the tag we searched for. We broke this by removing [this line from the end of our method](https://github.com/forem/forem/pull/13326/files#diff-27e95a7b175a5d22cf5a9a6423c4d4cb40ad7b6813d8686e32975966b88e81a8L27) and adding it to our if/else block. The failure is telling us that even when we are filtering by time_ago we need to also be filtering by the tags. This means that we need to ALWAYS add the tag filter BUT rather than doing it after the entire if/else block like before bc that didn't work, I think the best solution is to put it before the entire if/else block. That way we always have the tag filter AND then can adjust the published and score filters depending on if articles are returned when we add them. 
```ruby
relation = relation.limit(count)
relation = tags.size == 1 ? relation.cached_tagged_with(tags.first) : relation.tagged_with(tags)
relation = if time_ago == "latest"
                   ...
                 elsif time_ago
```

Does that all make sense?

Reply:
🎉 Ahh, yes! That all makes sense and aligns without what I'm seeing spec failure-wise. 😊